homeassistant/components/smartthings/sensor.py

```diff
@@ -567,59 +733,13 @@ async def async_setup_entry(
 ) -> None:
     """Add sensors for a config entry."""
     broker = hass.data[DOMAIN][DATA_BROKERS][config_entry.entry_id]
-    entities: list[SensorEntity] = []
-    for device in broker.devices.values():
-        for capability in broker.get_assigned(device.device_id, "sensor"):
-            if capability == Capability.three_axis:
-                entities.extend(
-                    [
-                        SmartThingsThreeAxisSensor(device, index)
-                        for index in range(len(THREE_AXIS_NAMES))
-                    ]
-                )
-            elif capability == Capability.power_consumption_report:
-                entities.extend(
-                    [
-                        SmartThingsPowerConsumptionSensor(device, report_name)
-                        for report_name in POWER_CONSUMPTION_REPORT_NAMES
-                    ]
-                )
-            else:
-                maps = CAPABILITY_TO_SENSORS[capability]
-                entities.extend(
-                    [
-                        SmartThingsSensor(
-                            device,
-                            m.attribute,
-                            m.name,
-                            m.default_unit,
-                            m.device_class,
-                            m.state_class,
-                            m.entity_category,
-                        )
-                        for m in maps
-                    ]
-                )
-
-        if broker.any_assigned(device.device_id, "switch"):
-            for capability in (Capability.energy_meter, Capability.power_meter):
-                maps = CAPABILITY_TO_SENSORS[capability]
-                entities.extend(
-                    [
-                        SmartThingsSensor(
-                            device,
-                            m.attribute,
-                            m.name,
-                            m.default_unit,
-                            m.device_class,
-                            m.state_class,
-                            m.entity_category,
-                        )
-                        for m in maps
-                    ]
-                )
```

Comment:
This PR also fixes a bug. This was added when the power/energy monitoring were moved from the switch entities to individual entities. BUT, not every switch has this. A switch in SmartThings means a device that can turn on and off. So, this code actually creates a power and energy sensor for every device, even if the device does not support those. If a device does support it, it will expose the energy meter and power meter capability, and be added like the rest of the sensors